homeassistant/components/samsungtv/config_flow.py

```diff
@@ -269,17 +263,17 @@ async def async_step_ssdp(
     ) -> data_entry_flow.FlowResult:
         """Handle a flow initialized by ssdp discovery."""
         LOGGER.debug("Samsung device found via SSDP: %s", discovery_info)
-        model_name: str = discovery_info.get(ATTR_UPNP_MODEL_NAME) or ""
-        self._udn = _strip_uuid(discovery_info[ATTR_UPNP_UDN])
-        if hostname := urlparse(discovery_info[ATTR_SSDP_LOCATION]).hostname:
+        model_name: str = discovery_info.upnp.get(ssdp.ATTR_UPNP_MODEL_NAME) or ""
+        self._udn = _strip_uuid(discovery_info.upnp[ssdp.ATTR_UPNP_UDN])
+        if hostname := urlparse(discovery_info.ssdp_location or "").hostname:
             self._host = hostname
         await self._async_set_unique_id_from_udn()
-        self._manufacturer = discovery_info[ATTR_UPNP_MANUFACTURER]
+        self._manufacturer = discovery_info.upnp[ssdp.ATTR_UPNP_MANUFACTURER]
         self._abort_if_manufacturer_is_not_samsung()
         if not await self._async_get_and_check_device_info():
             # If we cannot get device info for an SSDP discovery
             # its likely a legacy tv.
-            self._name = self._title = self._model = model_name
+            self._name = self._title = self._model = model_name  # pragma: no cover
```

Comment:
Added `# pragma: no cover` to ensure 100% coverage.
According to discord conversation, this was discussed between @chemelli74 and @bdraco.